python-checks/src/test/resources/checks/unusedLocalVariable.py

```diff
@@ -15,3 +15,9 @@ def using_locals(a, b):
   c = a + b
   # "locals" will include the "c" value
   return locals()
+
+def string_interpolation():
+    value1 = 1
+    value2 = 2 # Noncompliant
+    value3 = 3 # false-negative, value3 is not used as a variable in the string interpolation, see SONARPY-245
+    return f'The value1 is {value1}, value2bis, value3'
```

Comment:
As you chose a non-trivial approach, you should also test some more advanced cases, e.g. complex expressions: ```print(f'{2 * x}')```